samples/app-ReporterPlus/ReporterPlus/Bots/ReporterPlusBot.cs

```diff
@@ -0,0 +1,217 @@
+            Constants.BlobContainerName = configuration["BlobContainerName"];
+            Constants.BlobConnectionString = configuration["BlobConnectionString"];
+        }
+
+        protected override async Task<InvokeResponse> OnInvokeActivityAsync(ITurnContext<IInvokeActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if (turnContext.Activity.Name == "composeExtension/fetchTask")
+            {
+                MessagingExtensionAction action = new MessagingExtensionAction();
+                var actionJson = JsonConvert.DeserializeObject<MessagingExtensionActionDeserializer>(turnContext.Activity.Value.ToString());
+                action.CommandId = actionJson.commandId;
+                var task = await this.OnTeamsMessagingExtensionFetchTaskAsync(turnContext, action, cancellationToken);
+                return CreateInvokeResponse(task);
+            }
+            else if (turnContext.Activity.Name == "composeExtension/submitAction")
+            {
+                MessagingExtensionAction action = new MessagingExtensionAction();
+                var task = await this.OnTeamsMessagingExtensionSubmitActionAsync(turnContext, action, cancellationToken);
+                return CreateInvokeResponse(task);
+            }
+            else if (turnContext.Activity.Name == "task/fetch")
+            {
+                var actionJson = JsonConvert.DeserializeObject<ActionBase>(turnContext.Activity.Value.ToString());
+                var task = await ReturnViewDetails(actionJson.data.reqId);
+                return CreateInvokeResponse(task);
+            }
+            else if (turnContext.Activity.Name == "adaptiveCard/action")
+            {
+                var data = JsonConvert.DeserializeObject<ActionType>(turnContext.Activity.Value.ToString());
+                string channel = turnContext.Activity.ChannelId;
+                string verb = data.action.verb;
+                var blobInfo = BlobHelper.GetBlob(data.action.data.reqId, verb).Result;
+                string cardJsonString;
+
+                if (verb == "Refresh" && blobInfo.status == "Pending")
+                {
+                    CardHelper.CreateAdaptiveCardAttachment(blobInfo.status, blobInfo, channel, out cardJsonString);
+                    var cardResponse = JObject.Parse(cardJsonString);
+                    var res = new AdaptiveCardInvokeResponse()
+                    {
+                        StatusCode = 200,
+                        Type = AdaptiveCard.ContentType,
+                        Value = cardResponse
+                    };
+                    return CreateInvokeResponse(res);
+                }
+                else
+                {
+                    var adaptiveCardAttachment = CardHelper.CreateAdaptiveCardAttachment(blobInfo.status, blobInfo, channel, out cardJsonString);
+                    UpdateCardInTeams(blobInfo, adaptiveCardAttachment);
```

Comment:
Can we make this async so that we don't wait for the bot edit to complete? This would help us to respond to the `adaptiveCard/action` invoke request faster.